Patch author:  Orson Teodoro <orsonteodoro@hotmail.com>
Date:  Fri Feb 11 07:09:44 PM PST 2022 (Unix time:  1644635384)
Patch status:  Testing

License header for patched file:

// Copyright 2014 The Souper Authors. All rights reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

diff -urp souper-69536e134478ae1d44c912c90c3db96ad06437c1.orig/lib/Extractor/KLEEBuilder.cpp souper-69536e134478ae1d44c912c90c3db96ad06437c1/lib/Extractor/KLEEBuilder.cpp
--- souper-69536e134478ae1d44c912c90c3db96ad06437c1.orig/lib/Extractor/KLEEBuilder.cpp	2022-02-09 00:59:28.101234350 -0800
+++ souper-69536e134478ae1d44c912c90c3db96ad06437c1/lib/Extractor/KLEEBuilder.cpp	2022-02-09 01:01:47.595660496 -0800
@@ -66,13 +66,14 @@ public:
                          Inst *Precondition, bool Negate, bool DropUB) override {
     std::string SMTStr;
     llvm::raw_string_ostream SMTSS(SMTStr);
-    ConstraintManager Manager;
+    ConstraintSet Set;				// oteodoro:  updated to latest klee head 2.3_pre
+    ConstraintManager Manager(Set);		// oteodoro:  updated to latest klee head 2.3_pre
     Inst *Cand = GetCandidateExprForReplacement(BPCs, PCs, Mapping, Precondition, Negate, DropUB);
     if (!Cand)
       return std::string();
     prepopulateExprMap(Cand);
     ref<Expr> E = get(Cand);
-    Query KQuery(Manager, E);
+    Query KQuery(Set, E);			// oteodoro:  updated to latest klee head 2.3_pre
     ExprSMTLIBPrinter Printer;
     Printer.setOutput(SMTSS);
     Printer.setQuery(KQuery);
